pycloudlib/azure/util.py

```diff
@@ -35,57 +32,25 @@ def get_client(resource, config_dict: dict):
         The client for the resource passed as parameter.
 
     """
-    if not util.subp("command -v az", shell=True, rcs=[0, 1, 127]).ok:
-        logger.debug(
-            "No azure-cli 'az' command found. Trying explicit config params"
-        )
-    else:
-        try:
-            cli_credential = AzureCliCredential()
-            subscription_id = config_dict.get("subscriptionId")
-            client = resource(cli_credential, subscription_id=subscription_id)
-            return client
-        except CLIError:
-            logger.debug(
-                "No valid azure-cli config found."
-                " Trying explicit config params"
-            )
-        except ClientAuthenticationError:
-            logger.debug(
-                "Authentication error: No valid azure-cli config found."
-                " Trying explicit config params"
-            )
-
     required_keys = frozenset(
         {"clientId", "clientSecret", "tenantId", "subscriptionId"}
     )
     missing_keys = required_keys.difference(set(config_dict.keys()))
     if missing_keys:
         raise CloudSetupError(
-            "No AZ cli config found, missing required keys: {}".format(
+            "Missing required Azure credentials: {}".format(
                 ", ".join(missing_keys)
             )
         )
 
-    parameters = {
```

Comment:
Instead of reverting if there's a problem, can we check Xenial to see if removing this works? AFAIK, Pro is still using pycloudlib test things on Xenial, so as long as it works there, we should be ok.

Reply:
The minimal Ubuntu release used as host in pro tests is focal: https://github.com/canonical/ubuntu-pro-client/blob/b71a935df6ea8bf23d5ec2988941f2037cba0846/.github/workflows/ci-integration.yaml#L75

I have tested this on a bionic host, fixing the azure-related dependencies to the minimum ones, and it works.